ruma-events/src/room/message.rs

```diff
@@ -157,8 +148,34 @@ impl MessageEventContent {
     }
 }
 
+impl<'de> de::Deserialize<'de> for MessageEventContent {
+    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+    where
+        D: de::Deserializer<'de>,
+    {
+        use MessageEventContent::*;
+        let json = Box::<RawJsonValue>::deserialize(deserializer)?;
```

Comment:
Now that I see it here, I wonder whether we this would work instead:

```suggestion
        let json = <&RawJsonValue>::deserialize(deserializer)?;
```

Reply:
`cargo test` failed with:

```
---- edit_deserialization_061 stdout ----
thread 'edit_deserialization_061' panicked at 'called `Result::unwrap()` on an
`Err` value: Error("invalid type: string
\"{\\\"body\\\":\\\"s/foo/bar\\\",\\\"m.new_content\\\":{\\\"body\\\":\\\"bar\\\"},\\\"m.relates_to\\\":{\\\"event_id\\\":\\\"$1598361704261elfgc:localhost\\\",\\\"rel_type\\\":\\\"m.replace\\\"},\\\"msgtype\\\":\\\"m.text\\\"}\",
expected raw value", line: 0, column: 0)',
ruma-events/tests/room_message.rs:148:59
```

Reply:
What a useless error message. I remember some issues with `serde_json::Value` deserialization, maybe this is that. That resolves this concern in any case.

Reply:
Yeah we have `BTreeMap<String, JsonValue>` in a few places. Let's go with that for now, if you want you can open a new issue about switching all these over at once.